src/js/utils/spec-helpers/spec-helpers.js

```diff
@@ -15,17 +17,12 @@ define([], function() {
      * is applied as arguments to func.
      * @param {function} func The function that actually expresses the logic of the test.
      */
-    var withData = function(data, func) {
-        /* jshint loopfunc:true */
-        for (var name in data) {
-            if (data.hasOwnProperty(name)) {
-                (function(name) {
-                    it(name, function() {
-                        func.apply(this, data[name]);
-                    });
-                })(name);
-            }
-        }
+    withData = function(data, func) {
+        Object.keys(data).forEach(function(key) {
```

Comment:
Note to reviewers: `for (... in ...)` is banned by Airbnb ESLint because a) it has performance issues in some browsers and b) it loops over _all_ enumerable properties, not just object keys (e.g., `toString`)

In ES6 we can use [`for (... of ...)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) to do what `for (... in ...)` _looks_ like it does.